### DISCUSSION TITLE
Option to copy and send a message in 1-2 clicks

### DISCUSSION THREAD

**Discussion by x9136** [Chatterino/chatterino2#4590] [Ideas] [answered]
Hello!

During the emotional moments of the streams, people copy and paste the messages of others. This happens frequently on major channels. It's fun, but it always requires extra work, like highlighting, copying, pasting, and sending. It's better to put less effort into expressing emotions. Now it takes 5 steps to repeat a message: Hover over the message → Right-click → Copy the message → Paste → Send.

**The best solution is to click 1 button**. In that case, you need 2 steps: Hover over the message → Send.

It's important that it be obvious. For example, if the button appears when hovering over the message, it's obvious. But if you first need to configure that the button appears, it's not obvious. If you need to set a hotkey to send the message on your own behalf after hovering over it, it's not obvious.

I think it's hard to develop because you've hidden similar options behind right-clicking. Then what to do if hovering isn't possible?

**Right-click on the message and select the "Send this message" option**. Obviously. I think it's easy to develop, but there will be random clicks and 1 step is added: Hover over the message → Right click → Send.

**Right click on the message and select the "Repeat this message" option**. Once clicked, the message will be inserted into the input box with focus. 2 steps are added: Hover over the message → Right-click → Insert → Send. Not much different from what's out there. Not worth developing, not much use. 

What do you think?

**Comment by pajlada** [comment, marked as answer]
You can already make a command like this: `{1+}` or `I'm repeating your message LUL : {1+}` and enable the "Show in message menu"

This will show up like this when you right-click a message 
![image](https://user-images.githubusercontent.com/962989/235475913-c2a7f408-afe6-4b0a-a1ab-ca375c2d1231.png)

and instantly send the message when you click it.
I imagine this is an acceptable amount of clicks.

**Comment by x9136** [comment]
@pajlada Nice! I'll use it if it stays that way. I hope most users know about this possibility, but I learned from you.

It's not really about clicks, but more about convenience and the amount of action. Solving through commands makes you learn what commands are possible, how to set them up, and aim carefully as you move through the drop-down menu to the desired option. It's a lot easier when there is a button right away or there is an option and you don't have to aim hard. Imagine how you perform these solutions to compare with the current one.

I often see on streams when people copy other people's messages, so I suggested to simplify this action significantly.

**Comment by x9136** [comment]
Is it possible to extend the command so that after a message is sent, it remains in the input field? That way it would be ready to be sent again, which is often needed.

Since saving messages in history is broken, the {+1} command is useless when you need to send someone else's message more than one time. It's easier to copy-paste-send a message immediately, which stays in history and can be repeated via hotkey. And {+1} is worse in that when you have something in the buffer, using {+1} periodically sends information from the buffer, not what you right-clicked on.

**Comment by x9136** [comment]
Closed due no updates.
